qiita_pet/handlers/api_proxy/artifact.py

```diff
@@ -211,9 +211,12 @@ def artifact_post_req(user_id, filepaths, artifact_type, name,
             return {'status': 'error',
                     'message': "Can't create artifact, no files provided."}
 
-        # TODO: It might be good to have an API call that wraps to a lower
-        # level method that creates this job.
-        command = Command.get_validator(artifact_type)
+        # This try/except will catch the case when the plugins are not
+        # activated so there is no Validate for the given artifact_type
```

Comment:
If you want to catch the missing validator I think using [hasattr](https://docs.python.org/3/library/functions.html#hasattr) would likely be better i.e.

```python
if not hasattr(Command, 'attribute'):
     return {'status': 'error', 'message': 'The plugin xxxx has not been initialized'}
```

Reply:
I don't think that's gonna work as Command.get_validator will raise an error:
```python
In [4]: Command.get_validator('bla')                                                                                                                                               
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/svn_programs/qiita/qiita_db/software.py in get_validator(cls, artifact_type)
    166             try:
--> 167                 res = qdb.sql_connection.TRN.execute_fetchlast()
    168             except IndexError:

~/svn_programs/qiita/qiita_db/sql_connection.py in wrapper(self, *args, **kwargs)
     49                 "invoked within the context manager.")
---> 50         return func(self, *args, **kwargs)
     51     return wrapper

~/svn_programs/qiita/qiita_db/sql_connection.py in execute_fetchlast(self)
    333         """
--> 334         return self.execute()[-1][0][0]
    335 

IndexError: list index out of range

During handling of the above exception, another exception occurred:

QiitaDBError                              Traceback (most recent call last)
<ipython-input-4-73541800be9d> in <module>
----> 1 Command.get_validator('bla')

~/svn_programs/qiita/qiita_db/software.py in get_validator(cls, artifact_type)
    169                 raise qdb.exceptions.QiitaDBError(
    170                     "There is no command to generate the Validate for "
--> 171                     "artifact type '%s'" % artifact_type)
    172 
    173             return cls(res)

QiitaDBError: There is no command to generate the Validate for artifact type 'bla'
```

Reply:
I see, thanks. Should we instead be catching the QiitaDBError instead? Also maybe explaining what could be a reasonable workaround? Alternatively if you think there's many more unexpected exceptions that might come from this block of code, then let's just leave as is.

Reply:
Good point! Replacing ...

Reply:
👍